### PR TITLE
feat: add rapsearch2

### PR DESCRIPTION
**Pull request by bcai2**
Adds:
- Support for Rapsearch 2, in the form of `toolchest.rapsearch()` and `toolchest.rapsearch2()`. (The former is an alias of the latter.)
- Integration test for Rapsearch 2 (runtime ~4 mins)

Modifies:
- Sanity checking bugfixes for tools where `output_is_directory=False`.

### CODE REVIEW

File: toolchest_client/tools/api.py [hunk 1]
```diff
@@ -378,6 +374,42 @@ def megahit(output_path=None, tool_args="", read_one=None, read_two=None, interl
...
 
+def rapsearch(inputs, output_path=None, database_name="rapsearch_seqscreen", database_version="1",
+              tool_args="", **kwargs):
+    """Runs Rapsearch 2 via Toolchest.
+    :param inputs: Path to a FASTA/FASTQ file that will be passed in as input.
+    :param output_path: (optional) Base path to where the output file(s) will be downloaded.
+    (Functions the same way as the "-o" tag for Rapsearch.)
+    :param tool_args: (optional) Additional arguments to be passed to Rapsearch 2.
+    :param database_name: (optional) Name of database to use for Rapsearch 2 alignment. Defaults to standard DB.
+    :param database_version: (optional) Version of database to use for Kraken 2 alignment. Defaults to 1.
```
Comment (lebovic) [inline toolchest_client/tools/api.py:385]: Typo: Kraken 2

File: toolchest_client/tools/rapsearch.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            output_path=output_path,
+            inputs=inputs,
+            min_inputs=1,
+            max_inputs=1,
+            database_name=database_name,
+            database_version=database_version,
+            parallel_enabled=False,
+            output_type=OutputType.GZ_TAR,
+            output_is_directory=False,
+            output_names=[f"{output_primary_name}.m8"],  # .aln output may be omitted with certain args
```
Comment (lebovic) [inline toolchest_client/tools/rapsearch.py:35]: Are we returning a log file as well?

File: toolchest_client/tools/tool.py [hunk 3]
```diff
@@ -247,7 +248,10 @@ def _postflight(self, output):
                         output_file_path = f"{self.output_path}/{output_name}"
                         sanity_check(output_file_path)
                 else:
-                    sanity_check(self.output_path)
+                    # TODO: verify that this works with absolute pathing
```
Comment (lebovic) [inline toolchest_client/tools/tool.py:251]: We should probably add tests for this check

Comment (lebovic) [reply toolchest_client/tools/tool.py:251]: Possibly beyond scope of this PR, though

File: tests/test_rapsearch.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    Tests rapsearch2 on SeqScreen DB
+    """
+
+    test_dir = "./test_rapsearch"
+    os.makedirs(f"{test_dir}", exist_ok=True)
+    output_file_path_aln = f"./{test_dir}/rapsearch2.aln"
+    output_file_path_m8 = f"./{test_dir}/rapsearch2.m8"
+
+    toolchest.rapsearch(
+        tool_args="-e 1e-9",
```
Comment (lebovic) [inline tests/test_rapsearch.py:24]: Oof what an arg

**Review by lebovic — APPROVED** [review]
LGTM else a couple nits and the rapsearch vs rapsearch2 naming question

File: toolchest_client/tools/rapsearch.py (hunk 2 above)
Comment (bcai2) [reply toolchest_client/tools/rapsearch.py:35]: Not at the moment, but we can output stdout to a log file.

File: toolchest_client/tools/tool.py (hunk 3 above)
Comment (bcai2) [reply toolchest_client/tools/tool.py:251]: I'd agree on both fronts; I'm saving this for the comprehensive pathing PR

**Pull request merged**